package NRTLogAgent;

message TPQConfig {
    required string Server = 1;
    optional uint32 Port = 2 [default = 2135];
    required string Topic = 3;
    required uint32 TvmClientId = 4;
    required uint32 SourceIds = 5;
}

message TLimitConfig {
    optional uint32 ItemsCount = 1;
    optional uint64 ItemsSizeInBytes = 2;
}

message TLocalStorageConfig {
    required string Directory = 1;
    required uint64 AvailableSpaceInBytes = 2;
    required uint32 SegmentSizeInBytes = 3;
}

message TMessageSenderConfig {
    required TPQConfig Queue = 1;
    optional TLimitConfig InflightLimit = 2;
}

message TAppEndpoint {
    required string QueueName = 1;
    required uint32 MaxItems = 2;
    required uint32 MaxItemSizeInBytes = 3;
}

message TPublicEndpoint {
    required uint32 Port = 1;
}

message TConfig {
    required TAppEndpoint AppEndpoint = 1;
    required TPublicEndpoint PublicEndpoint = 2;
    required TLocalStorageConfig LocalStorage = 3;
    required TMessageSenderConfig MessageSender = 4;
}
